src/python/py_colorconfig.cpp

```diff
@@ -28,10 +28,9 @@
   (This is the Modified BSD License)
 */
 
-#include <utility>
-
 #include "py_oiio.h"
 #include <OpenImageIO/color.h>
+#include <utility>
```

Comment:
I'm not sure I understand what the issue here is.

Python.h, which is included by py_oiio.h, sets that macro and... what does that have to do with <utility>?

Reply:
Ugh, damn GitHub formatting...

what does that have to do with `<utility>`?

Reply:
Python.h sets **POSIX_C_SOURCE**, but it's unconditional and the same as what Python was compiled with.

`<utility>` can wind up included std-c headers that also set `POSIX_C_SOURCE`, but those tend to be conditional (adapting to incoming values).

So if Python.h is first, it's `POSIX_C_SOURCE` is used later in `<utlity>` and no redefinition occurs.
When `<utility>` is first, it chooses `POSIX_C_SOURCE` however it wants, which may conflict with `Python.h`

Reply:
This seems very brittle with respect to future additions of #include which most people think doesn't matter the order.

Perhaps a cleaner solution is to put this in py_oiio.h:

    #ifdef POSIX_C_SOURCE
        #undef POSIX_C_SOURCE
    #endif

Reply:
But with a comment:

    // It's bad news if POSIX_C_SOURCE is already defined with Python.h is included
    #ifdef POSIX_C_SOURCE
        #undef POSIX_C_SOURCE
    #endif 
    #include <Python.h>

Reply:
The Python guys seem pretty adamant their POSIX_C_SOURCE should rule for any modules compiled with it...whether that's valid or not...

My only thought is that doing the undef is making a new issue (albeit rare) where one might need to define `POSIX_C_SOURCE` rather keeping the it squarely in Python's court?

That said, don't have that strong of an opinion, I just though the comment below was pretty much because of this issue and the <utility> just wound up getting misplaced: https://github.com/OpenImageIO/oiio/blob/master/src/python/py_oiio.h#L14

Reply:
How about, in py_oiio.h :  

    #ifdef POSIX_C_SOURCE
    #  error "You must include Python.h (and therefore py_oiio.h) BEFORE anything that defines POSIX_C_SOURCE"
    #endif 

That will ensure that if somebody working on OIIO python bindings years from now changes the order of include files, they'll see this message and know what to do.

Reply:
LGTM (and certainly explicit!)
Should I add it on top ?

Reply:
Yeah, just put in py_oiio.h, right before including python.h, as an extra safety measure.

Reply:
In addition to your original reordering, I mean.